src/member-api.js

```diff
@@ -187,26 +187,47 @@ export class MemberApi extends TypedEmitter {
         filter: ([peerId, inviteResponse]) =>
           timingSafeEqual(peerId, deviceId) &&
           timingSafeEqual(invite.inviteId, inviteResponse.inviteId),
-        signal: abortController.signal,
+        signal: abortSignalAny([abortController.signal, signal]),
       }).then((args) => args?.[1])
 
     responsePromise.catch(noop)
 
+    signal.addEventListener(
+      'abort',
+      () => {
+        this.#rpc
+          .sendInviteCancel(deviceId, { inviteId: invite.inviteId })
+          .catch(noop)
+      },
+      { once: true }
+    )
+
     try {
       await this.#rpc.sendInvite(deviceId, invite)
       return await responsePromise
     } catch (err) {
       if (err instanceof Error && err.name === 'AbortError') {
-        throw new Error('Invite timed out')
+        throw new Error('Invite aborted')
       } else {
         throw err
       }
     } finally {
-      clearTimeout(timeoutId)
       abortController.abort()
     }
   }
 
+  /**
+   * Attempt to cancel an outbound invite, if it exists.
+   *
+   * No-op if we weren't inviting this device.
+   *
+   * @param {string} deviceId
+   * @returns {void}
+   */
+  requestCancelInvite(deviceId) {
```

Comment:
Suggestion: return a boolean indicating if the invite has been cancelled. Because an inprogress invite can be non-cancellable in its final stages, there is a window of time before `invite()` resolves when a user might call `requestCancelInvite()` and the invite might still resolve - having the boolean response might help client/front-end implementation.

Reply:
I like this idea but don't think we should implement it unless the frontend needs it.